### ISSUE TITLE
Link null after update to 5.5.3

### ISSUE THREAD

**Issue by AndiKeiser** [pimcore/pimcore#3543]
# Bug Report
 
 
### Expected behavior 
  
Working links in objects after update from 5.2.1 (219) to 5.5.3

### Actual behavior  
Links in objects are NULL after an update from 5.2.1 to 5.5.3

Working link in 5.2.1 looks like this:

`O:34:"Pimcore\Model\DataObject\Data\Link":14:{s:4:"text";s:8:"LocalFoo";s:12:"internalType";s:8:"document";s:8:"internal";i:1;s:6:"direct";N;s:8:"linktype";s:8:"internal";s:6:"target";s:6:"_blank";s:10:"parameters";s:0:"";s:6:"anchor";s:0:"";s:5:"title";s:8:"LocalFoo";s:9:"accesskey";s:0:"";s:3:"rel";s:0:"";s:8:"tabindex";s:0:"";s:5:"class";s:0:"";s:10:"attributes";s:0:"";}
  `

in 5.5.3

`O:34:"Pimcore\Model\DataObject\Data\Link":17:{s:7:"�*�text";s:8:"LocalFoo";s:15:"�*�internalType";s:8:"document";s:11:"�*�internal";i:1;s:9:"�*�direct";N;s:11:"�*�linktype";s:8:"internal";s:9:"�*�target";s:6:"_blank";s:13:"�*�parameters";s:0:"";s:9:"�*�anchor";s:0:"";s:8:"�*�title";s:8:"LocalFoo";s:12:"�*�accesskey";s:0:"";s:6:"�*�rel";s:0:"";s:11:"�*�tabindex";s:0:"";s:8:"�*�class";s:0:"";s:13:"�*�attributes";s:0:"";s:9:"�*�_owner";N;s:13:"�*�_fieldname";N;s:12:"�*�_language";N;}`

**Comment by brusch** [comment]
I'm not able to reproduce the problem. 
Anyone else with similar issues?

**Comment by aarongerig** [comment]
Having the same issue with Pimcore 5.5.3. I tried to debug it and it seems that the [PHP serialize function](https://github.com/pimcore/pimcore/blob/master/lib/Tool/Serialize.php#L33) causes that issue. Not sure why though...

**Comment by brusch** [comment]
I was able to reproduce the problem and also I've found the cause for it. 

Consider the following serialized `Link` object: 
```
O:34:"Pimcore\Model\DataObject\Data\Link":14:{s:4:"text";s:29:"www.oberschwaben-tourismus.de";s:12:"internalType";N;s:8:"internal";N;s:6:"direct";s:85:"http://www.oberschwaben-tourismus.de/reisethemen/kultur-barock/schloesser-burgen.html";s:8:"linktype";s:6:"direct";s:6:"target";s:6:"_blank";s:10:"parameters";s:0:"";s:6:"anchor";s:0:"";s:5:"title";s:0:"";s:9:"accesskey";s:0:"";s:3:"rel";s:0:"";s:8:"tabindex";s:0:"";s:5:"class";s:0:"";s:10:"attributes";s:0:"";}
```

In PHP 7.1 this looks like the following when unserialized: 
```
object(Pimcore\Model\DataObject\Data\Link)[1075]
  protected 'text' => null
  protected 'internalType' => null
  protected 'internal' => null
  protected 'direct' => null
  protected 'linktype' => null
  protected 'target' => null
  protected 'parameters' => null
  protected 'anchor' => null
  protected 'title' => null
  protected 'accesskey' => null
  protected 'rel' => null
  protected 'tabindex' => null
  protected 'class' => null
  protected 'attributes' => null
  protected '_owner' => null
  protected '_fieldname' => null
  protected '_language' => null
  public 'text' => string 'www.oberschwaben-tourismus.de' (length=29)
  public 'internalType' => null
  public 'internal' => null
  public 'direct' => string 'http://www.oberschwaben-tourismus.de/reisethemen/kultur-barock/schloesser-burgen.html' (length=85)
  public 'linktype' => string 'direct' (length=6)
  public 'target' => string '_blank' (length=6)
  public 'parameters' => string '' (length=0)
  public 'anchor' => string '' (length=0)
  public 'title' => string '' (length=0)
  public 'accesskey' => string '' (length=0)
  public 'rel' => string '' (length=0)
  public 'tabindex' => string '' (length=0)
  public 'class' => string '' (length=0)
  public 'attributes' => string '' (length=0)
  public 'path' => null
```

... and this is how it looks in PHP 7.2+ 
```
object(Pimcore\Model\DataObject\Data\Link)[1160]
  protected 'text' => string 'www.oberschwaben-tourismus.de' (length=29)
  protected 'internalType' => null
  protected 'internal' => null
  protected 'direct' => string 'http://www.oberschwaben-tourismus.de/reisethemen/kultur-barock/schloesser-burgen.html' (length=85)
  protected 'linktype' => string 'direct' (length=6)
  protected 'target' => string '_blank' (length=6)
  protected 'parameters' => string '' (length=0)
  protected 'anchor' => string '' (length=0)
  protected 'title' => string '' (length=0)
  protected 'accesskey' => string '' (length=0)
  protected 'rel' => string '' (length=0)
  protected 'tabindex' => string '' (length=0)
  protected 'class' => string '' (length=0)
  protected 'attributes' => string '' (length=0)
  protected '_owner' => null
  protected '_fieldname' => null
  protected '_language' => null
  public 'path' => string 'http://www.oberschwaben-tourismus.de/reisethemen/kultur-barock/schloesser-burgen.html' (length=85)
```

So it seems that PHP 7.1 does differentiate between public/private properties in the deserializing process, while PHP 7.2+ does not. 

Easiest fix imho is to upgrade to PHP 7.2+ if you're encountering this problem.
